src/app/store/reducers/error.reducers.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { Action } from '@ngrx/store';
+
+import { SetErrorMessage, ClearErrorMessage } from '../actions/Error.actions';
+import { ErrorReducer, ErrorState } from './error.reducers';
+
+describe('ErrorReducer', () => {
+  let state: ErrorState;
+  let resultState: ErrorState;
+  let action: any;
+
+  describe('when initialised', () => {
+    beforeEach(() => {
+      action = {} as Action;
+      const initState: ErrorState = { errors: [] };
+      resultState = ErrorReducer(initState, action);
+    });
+
+    it('should set the default state', () => {
+      expect(resultState).toMatchSnapshot();
```

Comment:
Do we want to have an expected state to assert on instead of using the snapshot to make sure we don't have false positive in the future?
Let's say if we agree that the app state is
```
error: {},
loading,
techRecord,
...
```
if we decide to change the error state to `error: {list: []}` the test will pass and didn't check whether we inadvertently modified the shape of the state or not.
I am not very familiar with snapShot approach on reducers so i would like to hear your thoughts on that please.
I would tend to define the expectedState shape like this instead of using snapShot to follow the "arrange, act, assert" pattern
```
// example for setting the error in the reducer
const action = {type: SetErrorMessage, payload: ['new error']}
const expectedState = {errors: ['new error'] }

expect(ErrorReducer(action, initialErrorState)).toMatch(expectedState)
```
With this approach above you can clearly read the effect (expectedState) it will have when the reducer receives the action. If we decide to change the state shape it will fail and it is good that we know so we can change it where it is necessary in the codebase.

Reply:
The generate snapshot should be the guide here and simplifies testing of reducers. So if any changes in shape occurs the test will fail